airflow/www/forms.py

```diff
@@ -194,9 +201,12 @@ class TaskInstanceEditForm(DynamicForm):
             ('up_for_retry', 'up_for_retry'),
         ),
         widget=Select2Widget(),
+        validators=[InputRequired()],
     )
     execution_date = DateTimeWithTimezoneField(
-        lazy_gettext('Execution Date'), widget=AirflowDateTimePickerROWidget()
+        lazy_gettext('Execution Date'),
+        widget=AirflowDateTimePickerROWidget(),
+        validators=[InputRequired()],
```

Comment:
This is a read only field so I don't think we need validators here, WDYT?

Reply:
I guess it's optional, but this means there's a nice red `*` so it's consistent.

In short: no harm, could leave or remove.